spring-boot/admin/src/main/resources/templates/admin/fragments/api-keys/edit.html

```diff
@@ -227,40 +236,37 @@ <h3> Select API user from table
                 // Add behaviour when the user saves the selection
                 $('#saveApiKeyButton').click(function () {
                     // Check if any github user selection
-                    if (table.row('.selected') !== null && table.row('.selected').data() !== null && typeof table.row('.selected').data() !== 'undefined')
-                    {
+                    if (table.row('.selected') !== null && table.row('.selected').data() !== null && typeof table.row('.selected').data() !== 'undefined') {
                         // Check if any service selection
-                        if ($("#selectedServiceId").val() !== null && $("#selectedServiceId").val() !== "" && typeof $("#selectedServiceId").val() !== 'undefined')
-                        {
-                            // Retrieve the github username which is common attribute in github user database and coatrack user database
-                            $("#selectedUserId").val(table.row('.selected').data().login);
+                        if ($("#selectedServiceId").val() !== null && $("#selectedServiceId").val() !== "" && typeof $("#selectedServiceId").val() !== 'undefined') {
+                            // Retrieve the github/keyrock username which is common attribute in github user database and coatrack user database
+                            ([[${@environment.getProperty('ygg.admin.idm.provider')}]].localeCompare("github") === 0) ? $("#selectedUserId").val(table.row('.selected').data().login) :
```

Comment:
I am wondering if we really need to differentiate here in the thymeleaf code which Idm provider is uses? Can't we get the `data()` and then take the attribute out via the key `[[${@environment.getProperty('oauth2.attributes.login.name')}]]` or similar? To be discussed.

Reply:
Create conditions only with the part that changes (which will be either **login** or **username**.

Reply:
I changed to code to get the `data()`, and then based on our global method to check if Github is being used idm, either use the **login** or **username**
Based on this, this is solved